homeassistant/components/vicare/climate.py

```diff
@@ -84,6 +84,12 @@
     HVAC_MODE_AUTO: VICARE_MODE_DHWANDHEATING,
 }
 
+HA_TO_VICARE_HVAC_HEATING_ALT = {
+    HVAC_MODE_HEAT: VICARE_MODE_HEATING,
+    HVAC_MODE_OFF: VICARE_MODE_OFF,
+    HVAC_MODE_AUTO: VICARE_MODE_HEATING,
```

Comment:
Viessmann calls the state "Heating" but actually it takes into account various conditions (outside temperature, room temperatures, etc) to decide if it actually heats or not.
This is not much different from the state VICARE_MODE_DHWANDHEATING that is used in  HA_TO_VICARE_HVAC_HEATING except for it taking Domestic Hot Water into account as well. A functionality some devices don't have.

Reply:
So in the above, the same modes are linked to auto & heat in Home Assistant. That is odd, ad there is thus no difference?

Reply:
Alternatively we could remove the heating mode completely in case it's not supported. Would that make sense to have a climate device that does not support this essential mode at all?

Reply:
I would remove 'auto' to mimic the approach from the device as much as possible so if the device call it heating then I suggest to call it heating in HA too.

Reply:
The original mapping used forcedNormal for hvac mode heating. The main difference from forcedNormal to heating is that it is **not** bound to any schedules the user set on his device (e.g. lower target temperature during night).
This makes a lot of sense because then the HA user can "force" his climate device into heating mode and override such a schedule.

If we cannot replicate the same behavior with the alternative mapping I would rather get rid of the hvac heating mode.

@hnykda can you please support us here?
This is related to the discussion in https://github.com/home-assistant/core/issues/63039
Is your Viessmann device respecting any schedules when it is in "heating" mode?

Reply:
Thanks everyone for trying to deal with this ViCare madness, really appreciate it :bow: and I am happy to help.

So, currently, my vicare app shows "normal" _program_ (according to my schedule) and API returns the same (cell n. 6):
```
In [4]: c = t.circuits[0]

In [5]: c.getActiveMode()
Out[5]: 'dhwAndHeating'

In [6]: c.getActiveProgram()
Out[6]: 'normal'

In [7]: c.getPrograms()
Out[7]: 
['active',
 'comfort',
 'comfortEnergySaving',
 'forcedLastFromSchedule',
 'noDemand',
 'normal',
 'normalEnergySaving',
 'reduced',
 'reducedEnergySaving',
 'standby',
 'summerEco']

In [8]: c.getModes()
Out[8]: ['standby', 'heating', 'dhw', 'dhwAndHeating']
```

![image](https://user-images.githubusercontent.com/2741256/154856440-f44438dd-8383-430c-90e4-c7d7c8ee6d37.png)
![image](https://user-images.githubusercontent.com/2741256/154856450-88dd3203-3e7b-4aff-98a6-b34d93fbdda5.png)

> Is your Viessmann device respecting any schedules when it is in "heating" mode?

Yes, I think your @oischinger assumption is correct: the `mode` `heating` or `dhwAndHeating` means in ViCare "heat according to the schedule" (just radiators, or radiators and hot water). According to schedule is indeed more complicated than just timelines - it's usually _also_ regulated by [either equitherm sensor](https://taphome.com/en/support/51281959) (that's my case, depends on the setting of Equitherm curve) or by a temperature sensor in e.g. a living room or wherever you have it.

In my case of Vitodens 100W, I can set three programs **only via the ViCare app and not via the API (have no idea why, it's simply not supported by the API)**: `reduced` `normal`  `comfort` and all four modes you see in the cell n.8 (that can be done via API ).

Reply:
Thanks for the details @hnykda

In that case I'd suggest the following.
- Only if vicare mode forcedNormal is available hvac heat shall be available.
- for hvac mode auto we set vicare mode "dhwAndHeating" and  "heating" as a fallback if that one is not available
- for hvac mode off we set vicare mode "forcedReduced" (this one does not turn of DHW heating) and if that's not available we set vicare mode "standby" which basically turns off the device.

As a result only in hvac mode "auto" the heating device's schedules will be respected. "off" and "heating" will override the schedule.

HVAC modes off and auto are always available to a user. HVAC mode heat might not be available for some devices.

Other thoughts?

Reply:
Thanks for the details @hnykda

In that case I'd suggest the following.
- Only if vicare mode forcedNormal is available hvac heat shall be available.
- for hvac mode auto we set vicare mode "dhwAndHeating" and  "heating" as a fallback if that one is not available
- for hvac mode off we set vicare mode "forcedReduced" (this one does not turn of DHW heating) and if that's not available we set vicare mode "standby" which basically turns off the device.

As a result only in hvac mode "auto" the heating device's schedules will be respected. "off" and "heating" will override the schedule.

HVAC modes off and auto are always available to a user. HVAC mode heat might not be available for some devices.

Other thoughts?

Reply:
Thanks for the details @hnykda

In that case I'd suggest the following.
- Only if vicare mode forcedNormal is available hvac heat shall be available.
- for hvac mode auto we set vicare mode "dhwAndHeating" and  "heating" as a fallback if that one is not available
- for hvac mode off we set vicare mode "forcedReduced" (this one does not turn of DHW heating) and if that's not available we set vicare mode "standby" which basically turns off the device.

As a result only in hvac mode "auto" the heating device's schedules will be respected. "off" and "heating" will override the schedule.

HVAC modes off and auto are always available to a user. HVAC mode heat might not be available for some devices.

Other thoughts?

Reply:
That sounds like the best solution to me (highly appreciated if this write up is in the docs).

Reply:
Other suggestion is using presets.
Then we could support hvac mode off for off and auto. With auto we then have the choice of all different presets viessmann offers.